### PR TITLE
optimize: remove Local implementation and replace with a call to Global

### PR DESCRIPTION
**Pull request by btracey**
This PR starts the process described in #482. It removes the existing Local implementation, replacing with a function that wraps Method to act as a GlobalMethod. This PR also adds a hack to fix an inconsistency with FunctionConverge between Global and Local (and a TODO to make it not a hack in the future). Finally it improves the Global documentation to help with shutdown.

### CODE REVIEW

File: optimize/local.go [hunk 1]
```diff
@@ -59,181 +56,176 @@ import (
...
-		if loc.Gradient != nil {
-			initG := settings.InitialGradient
-			if initG == nil {
+func (l *localGlobal) getStartingLocation(operation chan<- GlobalTask, result <-chan GlobalTask, task GlobalTask) Operation {
+	copy(task.X, l.InitX)
+	if l.Settings.UseInitialData {
+		task.F = l.Settings.InitialValue
+		if task.Gradient != nil {
+			g := l.Settings.InitialGradient
+			if task.Gradient != nil && g == nil {
```
Comment (vladimir-ch) [inline optimize/local.go]: task.Gradient is already checked for nil above.

File: optimize/unconstrained_test.go [hunk 2]
```diff
@@ -1182,36 +1182,36 @@ func testLocal(t *testing.T, tests []unconstrainedTest, method Method) {
 
 		result, err := Local(test.p, test.x, settings, method)
 		if err != nil {
-			t.Errorf("error finding minimum (%v) for:\n%v", err, test)
+			t.Errorf("Cas %d: error finding minimum (%v) for:\n%v", cas, err, test)
```
Comment (vladimir-ch) [inline optimize/unconstrained_test.go]: Why not spell these out as "Case"?

File: optimize/local.go (hunk 1 above)
Comment (btracey) [reply optimize/local.go]: Done.

File: optimize/unconstrained_test.go (hunk 2 above)
Comment (btracey) [reply optimize/unconstrained_test.go]: Done.

**Pull request merged**